iOS/MobileStarter/react-app/src/IModelPicker.tsx

```diff
@@ -0,0 +1,142 @@
+}
+
+interface IModelButtonProps extends Omit<ButtonProps, "title"> {
+  modelInfo: IModelInfo;
+  onCacheDeleted?: (modelInfo: IModelInfo) => void;
+};
+
+function IModelButton(props: IModelButtonProps) {
+  const { modelInfo, onClick, onCacheDeleted } = props;
+  const { minimalIModel, briefcase } = modelInfo;
+  const isMountedRef = useIsMountedRef();
+
+  const getTitle = (title: string, briefcase: LocalBriefcaseProps | undefined) => {
+    if (!briefcase) return title;
+    return title + " (" + fileSizeString(briefcase.fileSize) + ")";
+  }
+
+  const deleteBriefcase = React.useCallback(async (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    e.stopPropagation();
+    if (!briefcase) return;
+    try {
+      await MobileCore.deleteCachedBriefcase(briefcase);
+      if (!isMountedRef.current) return;
+      onCacheDeleted?.(modelInfo);
+    } catch (error) {
+      // There was a problem deleting the cached briefcase. Show the error.
+      presentError("DeleteErrorFormat", error, "HubScreen");
+    }
+  }, [briefcase, isMountedRef, modelInfo, onCacheDeleted]);
+
+  return <HubScreenButton
+    title={getTitle(minimalIModel.displayName, briefcase)}
+    onClick={onClick}
+  >
+    {briefcase && <div className="delete-button" onClick={deleteBriefcase}>
+      <IconImage iconSpec="icon-delete" />
+    </div>}
+  </HubScreenButton>
+}
+
+interface IModelListProps extends HubScreenButtonListProps {
+  models: IModelInfo[];
+  onClick?: (model: IModelInfo) => void;
+  onCacheDeleted?: (modelInfo: IModelInfo) => void;
+}
+
+function IModelList(props: IModelListProps) {
+  const { models, loading, onClick, onCacheDeleted, children } = props;
+  return <HubScreenButtonList loading={loading}>
+    {models.map((model, index) => <IModelButton key={index} modelInfo={model} onClick={() => onClick?.(model)} onCacheDeleted={onCacheDeleted} />)}
+    {children}
+  </HubScreenButtonList>;
+}
+
+async function getIModels(project: Project) {
+  const imodelsClient = new IModelsClient();
+  const accessToken = await IModelApp.getAccessToken();
+  const minimalIModels: MinimalIModel[] = [];
+  // Fetch the list of iModels.
+  for await (const minimalIModel of imodelsClient.iModels.getMinimalList({
+    authorization: AccessTokenAdapter.toAuthorizationCallback(accessToken),
+    urlParams: {
+      projectId: project.id,
+    }
+  })) {
+    minimalIModels.push(minimalIModel);
+  }
+  // Sort them by name using case-insensitive sort.
+  minimalIModels.sort((a, b) => a.displayName.localeCompare(b.displayName, undefined, { sensitivity: "base" }));
+  const iModelInfos: IModelInfo[] = [];
+  for (const minimalIModel of minimalIModels) {
+    const localBriefcases = await NativeApp.getCachedBriefcases(minimalIModel.id);
+    const briefcase = localBriefcases.length > 0 ? localBriefcases[0] : undefined;
+    iModelInfos.push({ minimalIModel, briefcase });
+  }
+  return iModelInfos;
+}
+
+export interface IModelPickerProps {
+  project: Project | undefined;
```

Comment:
Is it really possible to display an IModelPicker when project is undefined? It seems like that shouldn't be possible.

Reply:
Agreed, made it required and removed the isSignedOn prop.